### PR TITLE
Added arch types to Docker and binary releases

### LINKED ISSUE #2218: Inconsistent arm tagging on v7.5.0 release
We clone oauth2-proxy releases to our own container registry, specifically for amd64 and arm64 manifests.
When trying to clone the 7.5.0 release our workflow ran into the following error:
```
Step 1/1 : FROM quay.io/oauth2-proxy/oauth2-proxy:v7.5.0
v7.5.0: Pulling from oauth2-proxy/oauth2-proxy
no matching manifest for linux/arm64/v8 in the manifest list entries
```
It looks like there's a slight difference in how the tags were generated, as seen below. (screenshots taken from the [quay repository page](https://quay.io/repository/oauth2-proxy/oauth2-proxy?tab=tags))

We used to rely on the image tag suffixed with `-arm64` in the past, but moved to using the combined manifest once it started support arm64 linux. If necessary we can go back to that approach.

## Expected Behavior

v7.4.0 had a manifest for `arm64`
image

## Current Behavior

v7.5.0 does not have a manifest for `arm64`, but rather two `arm` manifests.
image

## Steps to Reproduce (for bugs)

1.  Start a terminal in a docker-enabled, arm64 linux machine.
2.  `echo 'FROM quay.io/oauth2-proxy/oauth2-proxy:v7.5.0' > /tmp/Dockerfile`
3.  `docker build -f /tmp/Dockerfile .`

### LINKED ISSUE #2219: Include darwin arm64 in releases
Thanks for the most recent release yesterday! Would be nice to include a release of oauth2-proxy for darwin arm64. Would this be possible?

### PR DESCRIPTION
**Pull request by kvanzuijlen**
## Description
Adds build outputs for darwin-arm64, linux-armv5 and linux-armv7

## Motivation and Context
Fixes #2219
Fixes #2218 

## How Has This Been Tested?
Ran `make release` locally. All release artifacts were generated succesfully.

## Checklist:

- [x] My change requires a change to the documentation or CHANGELOG.
- [x] I have updated the documentation/CHANGELOG accordingly.
- [x] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -49,35 +49,27 @@ DOCKER_BUILDX_PUSH_X_PLATFORM := $(DOCKER_BUILDX_PUSH) --platform ${DOCKER_BUILD
 
 .PHONY: docker
 docker:
-	$(DOCKER_BUILDX_X_PLATFORM) -f Dockerfile -t $(REGISTRY)/oauth2-proxy:latest .
+	$(DOCKER_BUILDX_X_PLATFORM) -f Dockerfile -t $(REGISTRY)/oauth2-proxy:latest -f Dockerfile -t $(REGISTRY)/oauth2-proxy:${VERSION} .
```
Comment (tuunit) [inline Makefile:52]:
```suggestion
	$(DOCKER_BUILDX_X_PLATFORM) -f Dockerfile -t $(REGISTRY)/oauth2-proxy:latest -t $(REGISTRY)/oauth2-proxy:${VERSION} .
```

File: Makefile [hunk 2]
```diff
@@ -49,35 +49,27 @@ DOCKER_BUILDX_PUSH_X_PLATFORM := $(DOCKER_BUILDX_PUSH) --platform ${DOCKER_BUILD
...
-	$(DOCKER_BUILDX) --platform linux/arm64 -t $(REGISTRY)/oauth2-proxy:${VERSION}-arm64 .
-	$(DOCKER_BUILDX) --platform linux/ppc64le -t $(REGISTRY)/oauth2-proxy:latest-ppc64le .
-	$(DOCKER_BUILDX) --platform linux/ppc64le -t $(REGISTRY)/oauth2-proxy:${VERSION}-ppc64le .
-	$(DOCKER_BUILDX) --platform linux/arm/v6 -t $(REGISTRY)/oauth2-proxy:latest-armv6 .
-	$(DOCKER_BUILDX) --platform linux/arm/v6 -t $(REGISTRY)/oauth2-proxy:${VERSION}-armv6 .
+	$(DOCKER_BUILDX) --platform linux/amd64 -t $(REGISTRY)/oauth2-proxy:latest-amd64 -t $(REGISTRY)/oauth2-proxy:${VERSION}-amd64 .
+	$(DOCKER_BUILDX) --platform linux/arm64 -t $(REGISTRY)/oauth2-proxy:latest-arm64 -t $(REGISTRY)/oauth2-proxy:${VERSION}-arm64 .
+	$(DOCKER_BUILDX) --platform linux/ppc64le -t $(REGISTRY)/oauth2-proxy:latest-ppc64le -t $(REGISTRY)/oauth2-proxy:${VERSION}-ppc64le .
+	$(DOCKER_BUILDX) --platform linux/arm/v6 -t $(REGISTRY)/oauth2-proxy:latest-armv6 -t $(REGISTRY)/oauth2-proxy:${VERSION}-armv6 .
+	$(DOCKER_BUILDX) --platform linux/arm/v7 -t $(REGISTRY)/oauth2-proxy:latest-armv7 -t $(REGISTRY)/oauth2-proxy:${VERSION}-armv7 .
```
Comment (tuunit) [inline Makefile:60]:
```suggestion
	$(DOCKER_BUILDX) --platform linux/amd64   -t $(REGISTRY)/oauth2-proxy:latest-amd64 -t $(REGISTRY)/oauth2-proxy:${VERSION}-amd64 .
	$(DOCKER_BUILDX) --platform linux/arm64   -t $(REGISTRY)/oauth2-proxy:latest-arm64 -t $(REGISTRY)/oauth2-proxy:${VERSION}-arm64 .
	$(DOCKER_BUILDX) --platform linux/ppc64le -t $(REGISTRY)/oauth2-proxy:latest-ppc64le -t $(REGISTRY)/oauth2-proxy:${VERSION}-ppc64le .
	$(DOCKER_BUILDX) --platform linux/arm/v6  -t $(REGISTRY)/oauth2-proxy:latest-armv6 -t $(REGISTRY)/oauth2-proxy:${VERSION}-armv6 .
	$(DOCKER_BUILDX) --platform linux/arm/v7  -t $(REGISTRY)/oauth2-proxy:latest-armv7 -t $(REGISTRY)/oauth2-proxy:${VERSION}-armv7 .
```

Comment (tuunit) [reply Makefile:60]: a bit of formatting

File: Makefile [hunk 3]
```diff
@@ -49,35 +49,27 @@ DOCKER_BUILDX_PUSH_X_PLATFORM := $(DOCKER_BUILDX_PUSH) --platform ${DOCKER_BUILD
...
-	$(DOCKER_BUILDX_PUSH) --platform linux/arm64 -t $(REGISTRY)/oauth2-proxy:${VERSION}-arm64 .
-	$(DOCKER_BUILDX_PUSH) --platform linux/ppc64le -t $(REGISTRY)/oauth2-proxy:latest-ppc64le .
-	$(DOCKER_BUILDX_PUSH) --platform linux/ppc64le -t $(REGISTRY)/oauth2-proxy:${VERSION}-ppc64le .
-	$(DOCKER_BUILDX_PUSH) --platform linux/arm/v6 -t $(REGISTRY)/oauth2-proxy:latest-armv6 .
-	$(DOCKER_BUILDX_PUSH) --platform linux/arm/v6 -t $(REGISTRY)/oauth2-proxy:${VERSION}-armv6 .
+	$(DOCKER_BUILDX_PUSH) --platform linux/amd64 -t $(REGISTRY)/oauth2-proxy:latest-amd64 -t $(REGISTRY)/oauth2-proxy:${VERSION}-amd64 .
+	$(DOCKER_BUILDX_PUSH) --platform linux/arm64 -t $(REGISTRY)/oauth2-proxy:latest-arm64 -t $(REGISTRY)/oauth2-proxy:${VERSION}-arm64 .
+	$(DOCKER_BUILDX_PUSH) --platform linux/ppc64le -t $(REGISTRY)/oauth2-proxy:latest-ppc64le -t $(REGISTRY)/oauth2-proxy:${VERSION}-ppc64le .
+	$(DOCKER_BUILDX_PUSH) --platform linux/arm/v6 -t $(REGISTRY)/oauth2-proxy:latest-armv6 -t $(REGISTRY)/oauth2-proxy:${VERSION}-armv6 .
+	$(DOCKER_BUILDX_PUSH) --platform linux/arm/v7 -t $(REGISTRY)/oauth2-proxy:latest-armv7 -t $(REGISTRY)/oauth2-proxy:${VERSION}-armv7 .
```
Comment (tuunit) [inline Makefile:72]:
```suggestion
	$(DOCKER_BUILDX_PUSH) --platform linux/amd64   -t $(REGISTRY)/oauth2-proxy:latest-amd64 -t $(REGISTRY)/oauth2-proxy:${VERSION}-amd64 .
	$(DOCKER_BUILDX_PUSH) --platform linux/arm64   -t $(REGISTRY)/oauth2-proxy:latest-arm64 -t $(REGISTRY)/oauth2-proxy:${VERSION}-arm64 .
	$(DOCKER_BUILDX_PUSH) --platform linux/ppc64le -t $(REGISTRY)/oauth2-proxy:latest-ppc64le -t $(REGISTRY)/oauth2-proxy:${VERSION}-ppc64le .
	$(DOCKER_BUILDX_PUSH) --platform linux/arm/v6  -t $(REGISTRY)/oauth2-proxy:latest-armv6 -t $(REGISTRY)/oauth2-proxy:${VERSION}-armv6 .
	$(DOCKER_BUILDX_PUSH) --platform linux/arm/v7  -t $(REGISTRY)/oauth2-proxy:latest-armv7 -t $(REGISTRY)/oauth2-proxy:${VERSION}-armv7 .
```

File: Makefile (hunk 1 above)
Comment (kvanzuijlen) [reply Makefile]: Thx @tuunit <3

**Comment by kvanzuijlen** [comment]
> shouldn't we keep armv7 and armv8? why did you remove armv8?

Assuming this is about the Makefile; I renamed it back to arm64 since that's likely what caused the issue described in #2218.

**Review by JoelSpeed — COMMENTED** [review]
With this change we are no longer pushing the `latest` tags for anything, I don't think that's a desired change is it?

**Comment by kvanzuijlen** [comment]
> With this change we are no longer pushing the `latest` tags for anything, I don't think that's a desired change is it?

@JoelSpeed I think that should still work, at least based on my own test:
image
I'll purge my previous pushes and try again just to be sure! I'll let you know

**Comment by kvanzuijlen** [comment]
@JoelSpeed 
still works:
image
(v7.5.0 + latest are also there)

Have tested it with the latest changes I just pushed

**Review by JoelSpeed — DISMISSED** [review]

**Review by JoelSpeed — APPROVED** [review]

**Pull request merged**